mlir-tensorrt/compiler/lib/Dialect/TensorRTRuntime/IR/TensorRTRuntime.cpp

```diff
@@ -113,6 +113,118 @@ void EnqueueOp::getEffects(
   }
 }
 
+//===----------------------------------------------------------------------===//
+// EnqueueAllocOp
+//===----------------------------------------------------------------------===//
+
+void EnqueueAllocOp::inferOperandKind(
+    ArrayRef<TensorKindLattice *> operands,
+    ArrayRef<const TensorKindLattice *> results,
+    llvm::function_ref<void(OpOperand &, TensorKind)> setOperandKind) {
+  for (OpOperand &operand : getOperation()->getOpOperands()) {
+    if (!isa<ShapedType>(operand.get().getType()))
+      continue;
+    if (isOperandOnHost(&operand)) {
+      setOperandKind(operand, TensorKind::Host);
+      continue;
+    }
+    setOperandKind(operand, TensorKind::Device);
+  }
+}
+
+TensorKind EnqueueAllocOp::getStaticOperandTensorKind(OpOperand &operand) {
+  return isOperandOnHost(&operand) ? TensorKind::Host
+                                                     : TensorKind::Device;
+}
+
+static bool hasCanonicalStrides(MemRefType memrefType) {
+  if (auto layoutAttr =
+          memrefType.getLayout().dyn_cast_or_null<StridedLayoutAttr>()) {
+    // Check if the strides are canonical
+    ArrayRef<int64_t> strides = layoutAttr.getStrides();
+    ArrayRef<int64_t> shape = memrefType.getShape();
+
+    int64_t expectedStride = 1;
+    for (int i = shape.size() - 1; i >= 0; --i) {
+      if (strides[i] != ShapedType::kDynamic && strides[i] != expectedStride)
+        return false;
+      if (shape[i] != ShapedType::kDynamic)
+        expectedStride *= shape[i];
+    }
+    return true;
+  }
+
+  // If there's no explicit layout, assume it's canonical
```

Comment:
Be conservative! This is not a conservative decision. You must always err on the side of correctness. Anyways, `memrefType.getLayout().isIdentity()` should take care of this.